CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+cmake_minimum_required(VERSION 3.23.1 FATAL_ERROR)
+
+enable_language(C CXX)
+
+include(fetch_rapids.cmake)
+
+include(rapids-cmake)
+include(rapids-cpm)
+include(rapids-find)
+
+rapids_cpm_init()
+
+rapids_find_package(
+  CUDAToolkit 12.2 REQUIRED
```

Comment:
We are quite likely to always want to build with the very latest CUDA, because the aim is to support PTX coming from any released version of the CUDA toolkit.

Reply:
So the trick is going to be determining what the latest CUDA to support in different distribution mechanisms is. For pip wheels it may be that we always want to use the latest released CUDA since that's always installable, whereas for conda it may be lagged based on when we update conda packages to support the latest CUDA. IOW I don't know if it makes sense to encode a specific minor version here, but rather rely on that being set correctly in the appropriate build scripts for different distributions (e.g. meta.yaml recipes for conda or containers used for pip wheel builds). We'll have to maintain that second set of sources anyway, so we may as well make those the source of truth.